utils/proc/process.go

```diff
@@ -209,6 +216,29 @@ func killChildren(ctx context.Context, p *process.Process) (err error) {
 
 }
 
+func isProcessRunning(p *process.Process) (running bool) {
+	if p == nil {
+		return
+	}
+	running, _ = p.IsRunning()
+	if !running {
+		return
+	}
+	// from man 2 kill: If sig is 0, then no signal is sent, but error checking is still performed; this can be used to check for the existence of a process ID or process group ID.
+	exist, _ := process.PidExists(p.Pid)
+	if !exist {
+		running = false
+		return
+	}
+	status, err := p.Status()
+	if err != nil {
+		return
```

Comment:
so this is because, we are trying to be more granular by checking the status. Nonetheless, on some platform such as Windows, the status is not implemented and so in this case we want to fallback to the IsRunning method, which is not thorough on unix platform as a zombie process is reported as running